src/mongo/client/dbclient_rs.cpp

```diff
@@ -426,10 +427,25 @@ namespace {
                     break;
                 }
 
-                conn->auth( params );
+                BSONObj interposedParams;
+
+                // Force usage of SCRAM-SHA-1 if host has maxWireVersion > 3.
+                if ((conn->getMaxWireVersion() >= 3) &&
+                    (params[saslCommandMechanismFieldName].str() == "MONGODB-CR")) {
+                    BSONObjBuilder interposedParamsBob;
+                    interposedParamsBob.append(saslCommandMechanismFieldName, "SCRAM-SHA-1");
+                    interposedParamsBob.appendElementsUnique(params);
+                    interposedParams = interposedParamsBob.obj();
+                }
+                else {
+                    interposedParams = params;
+                }
+
+                conn->auth( interposedParams );
 
                 // Cache the new auth information since we now validated it's good
-                _auths[params[saslCommandUserDBFieldName].str()] = params.getOwned();
+                _auths[interposedParams[saslCommandUserDBFieldName].str()] =
```

Comment:
I know you just changed params to interposedParams here, correctly. However, this code looks a little strange. If params/interposedParams here doesn't have a saslCommandUserDBFieldName, then this will store an entry in _auths with a key of ""?